docs/query-builder.txt

```diff
@@ -1035,6 +1035,55 @@ following MongoDB-specific write operations:
 Upsert a Document Example
 ~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Starting in v4.7, you can perform an upsert operation by using either of
+the following query builder methods:
+
+- ``upsert()``: When you use this method, you can perform a **batch
+  upsert** to change or insert multiple documents in one operation.
+
+- ``update()``: When you use this method, you must specify the
+  ``upsert`` option to update all documents that match the query filter
+  or insert one document if no documents are matched. Only this upsert method
+  is supported in versions v4.6 and earlier.
+
+Upsert Method
+^^^^^^^^^^^^^
+
+The ``upsert()`` query builder method accepts the following parameters:
+
+- Array of fields and values that specify documents to update or insert.
+- Array of fields that uniquely identify documents in your first array parameter.
+- Array of fields to update if a matching document exists. If you omit
+  this parameter, {+odm-short+} updates all fields.
+
+The following example shows how to use the ``upsert()`` query builder method
+to update or insert documents based on the following instructions:
+
+- Specify a document in which the value of the ``title`` field is
+  ``'Inspector Maigret'``, the value of the ``recommended`` field is ``false``,
+  and the value of the ``runtime`` field is ``128``.
+
+- Specify a document in which the value of the ``title`` field is
+  ``'Petit Maman'``, the value of the ``recommended`` field is
+  ``true``, and the value of the ``runtime`` field is ``72``.
+
+- Indicate that the ``title`` field uniquely identifies documents in the
+  scope of your operation.
+
+- Update only the ``recommended`` field in matched documents.
+
+.. literalinclude:: /includes/query-builder/QueryBuilderTest.php
+   :language: php
+   :dedent:
+   :start-after: begin upsert
+   :end-before: end upsert
+
+The ``upsert()`` query builder method returns the number of
+documents that the operation updated, inserted, and modified.
+
```

Comment:
You can add a note that events are not triggered by this operation. If events are necessary, use `createOrFirst`.